pkg/controllers/networking/pod_controller.go

```diff
@@ -111,7 +111,35 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result
 	}
 
 	if pod.DeletionTimestamp != nil {
-		if strategy.OwnByStatefulWorkload(pod) {
+		var ownedObj client.Object = pod
+
+		// For terminating pods with no controller owner reference, try to get
+		// owner reference from ip instance.
+		if metav1.GetControllerOf(pod) == nil {
+			var ipInstanceList = &networkingv1.IPInstanceList{}
+			if err = r.List(ctx, ipInstanceList,
+				client.MatchingLabels{constants.LabelPod: pod.Name},
+				client.InNamespace(pod.Namespace),
+			); err != nil {
+				return ctrl.Result{}, wrapError("failed to list ip instance for pod", err)
+			}
+			for i := range ipInstanceList.Items {
+				ipInstance := &ipInstanceList.Items[i]
+				if !ipInstance.DeletionTimestamp.IsZero() {
+					continue
+				}
+				if metav1.GetControllerOf(ipInstance) != nil {
+					ownedObj = ipInstance.DeepCopy()
+					break
+				}
+			}
+		}
+		// If we cannot find owner ref on pod nor ip instance, just remove pod finalizer.
+		if metav1.GetControllerOf(ownedObj) == nil {
```

Comment:
this judgement can only be false when `metav1.GetControllerOf(pod) == nil`, so these three lines should be moved inside the judgement section above.